application/src/main/java/org/togetherjava/tjbot/commands/mathcommands/TeXCommand.java

```diff
@@ -92,6 +100,21 @@ public void onSlashCommand(@NotNull final SlashCommandEvent event) {
             .queue();
     }
 
+    @NotNull
+    private String convertInlineLatexToFull(String latex) {
+        Matcher matcher = INLINE_LATEX_REPLACEMENT.matcher(latex);
+        StringBuilder sb = new StringBuilder();
+        while (matcher.find()) {
+            boolean isInsideMathRegion = matcher.group(1) != null;
+            if (isInsideMathRegion) {
+                sb.append(matcher.group(1).replace("$", ""));
+            } else {
+                sb.append("\\text{").append(matcher.group(2)).append("}");
+            }
+        }
+        return sb.toString();
+    }
```

Comment:
A quick comment at the beginning of the method, or as Javadoc could help. Sth along the lines of:
```java
// Converts inline latex like 'hello $\frac{x}{2}$ world'
// to full latex '\text{hello }\frac{x}{2}\text{ world}'.
```